### ISSUE TITLE
Restore problem when work with multiple tf.contrib.lookup.MutableHashTable

### ISSUE THREAD

**Issue by wrongtest** [tensorflow/tensorflow#19528]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: 
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 3.10.0-693.2.2.el7.x86_64
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.8.0
- **Python version**: Python 2.7.14 :: Anaconda
- **Bazel version (if compiling from source)**: None
- **GCC/Compiler version (if compiling from source)**: None
- **CUDA/cuDNN version**: None
- **GPU model and memory**: None
- **Exact command to reproduce**: See below

### Describe the problem
We try to use multiple tf.contrib.lookup.MutableHashTable objects in the script. But after do checkpoint with Saver.save(), we find that it can not restore table contents successfully. 

The problem disappears if we explicitly specify name field for tables. It seems that the MutableHashTable save contents with spec key "xxxx-keys" and "xxxx-values", where "xxxx" is the object's name field defaults to "MutableHashTable".

https://github.com/tensorflow/tensorflow/blob/93bc2e2072e0daccbcff7a90d397b704a9e8f778/tensorflow/contrib/lookup/lookup_ops.py#L463-L467

Maybe it is better to associate unique name for each table object if not explicitly specified?

### Source code / logs
Create and insert tables, and save
```python
keys = tf.placeholder(dtype=tf.string, shape=[None])
values = tf.placeholder(dtype=tf.int64, shape=[None])
table1 = tf.contrib.lookup.MutableHashTable(tf.string, tf.int64, -1)
table2 = tf.contrib.lookup.MutableHashTable(tf.string, tf.int64, -1)
insert_table1 = table1.insert(keys, values)
insert_table2 = table2.insert(keys, values)
saver = tf.train.Saver()
with tf.Session() as sess:
    sess.run(insert_table1, feed_dict={keys: ["a"], values: [1]})
    sess.run(insert_table2, feed_dict={keys: ["b"], values: [2]})
    print "table1:", sess.run(table1.export())
    print "table2:", sess.run(table2.export())
    saver.save(sess, "checkpoint/test")
```
table1: (array(['a'], dtype=object), array([1]))
table2: (array(['b'], dtype=object), array([2]))

Try restore contents
```python
table1 = tf.contrib.lookup.MutableHashTable(tf.string, tf.int64, -1)
table2 = tf.contrib.lookup.MutableHashTable(tf.string, tf.int64, -1)
saver = tf.train.Saver()
with tf.Session() as sess:
    saver.restore(sess, "checkpoint/test")
    print "table1:", sess.run(table1.export())
    print "table2:", sess.run(table2.export())
```
table1: (array([], dtype=object), array([], dtype=int64))
table2: (array(['b'], dtype=object), array([2]))

**Comment by labyrinth1992** [comment]
exactly same problem..

**Comment by tobegit3hub** [comment]
Since `Operator` and `Variable` has handled the conflict of the default names, it would better to manage the allocated names by TensorFlow.

**Comment by angersson** [comment]
Thanks for providing clear code and explanation!

@ysuematsu Can you take a look at this?

**Comment by tensorflowbutler** [comment]
Nagging Assignees @ysuematsu, @angersson: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @ysuematsu, @angersson: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tobegit3hub** [comment]
Any update of this issue? It would be great if we can make the contribution to fix that by automatically assigning unique names for Table operators.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @ysuematsu, @angersson: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @ysuematsu, @angersson: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ysuematsu: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ysuematsu: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ysuematsu: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ysuematsu: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ysuematsu: It has been 89 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ysuematsu: It has been 104 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ysuematsu: It has been 119 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mohantym** [comment]
We see that you are using old version of Tensorflow which is officially considered as end of life, We recommend that you upgrade to 2.4 or later version and let us know if the issue still persists in newer versions .Please open a new issue in case you face any errors, we will get you the right help .Thanks!
